### PR TITLE
Forbid class concatenation

### LINKED ISSUE #1: Should we compile the scss?
Because we do not compile the SCSS files, we can not detect BEM errors like this one:

```scss
.block__element {
    &__antoher-element {}
}
```

Another example is a selector created by a mixin.

We could compile the scss, but we will lose the source references like the line number :disappointed: 

What should we do?

### PR DESCRIPTION
**Pull request by tzi**
Avoid the main source of the #1 issue, waiting for a better solution.

 * [x] Development
 * [x] Tests

Error example:

```scss
.block {

  /* OK */
  &::after { }
  & .block__close { }

  /* ERROR */ 
  &--big { } 
  &__close {}
}
```

### CODE REVIEW

File: packages/bemlinter/__tests__/sources/class-concat.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    right: 0;
+    bottom: 0;
+    background: hotpink;
+  }
+
+  &__close {
+    position: absolute;
+    top: 14px;
+    right: 15px;
+
```
Comment (Osirisxxl) [inline packages/bemlinter/__tests__/sources/class-concat.scss]: this blank line seems useless

File: packages/bemlinter/src/lint.js [hunk 2]
```diff
@@ -125,4 +130,13 @@ module.exports = (sources, userOptions = defaultOptions) => {
       }
     });
   }
+  
+  function checkSelector($, filePath, blockName) {
+    eachWrapper($('selector'), wrapper => {
+      const selector = nodeToString(wrapper.node).trim();
+      if (/&[-_a-z]/.test(selector)) {
```
Comment (Osirisxxl) [inline packages/bemlinter/src/lint.js]: I think that, given the pattern you are testing here, you could add a test case to fail on `&foo { //... }`

Comment (Osirisxxl) [inline packages/bemlinter/src/lint.js]: Also from a top level point of view, I'd rather use an exclusive pattern or deal with it find & occurrences and checking next char or object.

File: packages/bemlinter/__tests__/sources/class-concat.scss (hunk 1 above)
Comment (busuttil) [inline packages/bemlinter/__tests__/sources/class-concat.scss]: i think is more readable

File: packages/bemlinter/__tests__/sources/class-concat.scss [hunk 3]
```diff
@@ -1,11 +1,14 @@
 .class-concat {
-  display: block;
-  position: absolute;
-  z-index: 10;
-  top: 50%;
-  transform: translateY(-50%);
-  width: 29px;
-  height: 29px;
+  
+  &{
```
Comment (tzi) [inline packages/bemlinter/__tests__/sources/class-concat.scss]: Pour avoir un test sans élément après `&`

File: packages/bemlinter/__tests__/__snapshots__/single-file.test.js.snap [hunk 4]
```diff
@@ -4,8 +4,8 @@ exports[`Bemlinter of alright.scss should lint without error 1`] = `"OK: bemlint
 
 exports[`Bemlinter of class-concat.scss should detect class concatenation 1`] = `
 "  ✗ class-concat
-[class-concat.scss:20] Error: \"&__close\" should not concatenate classes.
-[class-concat.scss:28] Error: \"&--big\" should not concatenate classes.
+[class-concat.scss:23] Error: \"&__close\" should not concatenate classes.
+[class-concat.scss:31] Error: \"&--big\" should not concatenate classes.
```
Comment (Osirisxxl) [inline packages/bemlinter/__tests__/__snapshots__/single-file.test.js.snap]: yay, I "love" snapshots...

**Pull request merged**